src/python/zquantum/qcbm/cost_function.py

```diff
@@ -1,14 +1,53 @@
-from zquantum.core.interfaces.functions import function_with_gradient, StoreArtifact
-from zquantum.core.interfaces.backend import QuantumBackend
-from zquantum.core.interfaces.ansatz import Ansatz
+from typing import Callable
+
+import numpy as np
 from zquantum.core.bitstring_distribution import (
     BitstringDistribution,
     evaluate_distribution_distance,
 )
-from zquantum.core.utils import ValueEstimate
 from zquantum.core.gradients import finite_differences_gradient
-from typing import Union, Callable
-import numpy as np
+from zquantum.core.interfaces.ansatz import Ansatz
+from zquantum.core.interfaces.backend import QuantumBackend
+from zquantum.core.interfaces.cost_function import CostFunction
+from zquantum.core.interfaces.functions import StoreArtifact, function_with_gradient
+from zquantum.core.utils import ValueEstimate
+
+
+def create_QCBM_cost_function(
+    ansatz: Ansatz,
+    backend: QuantumBackend,
+    n_samples: int,
+    distance_measure: Callable,
+    distance_measure_parameters: dict,
+    target_bitstring_distribution: BitstringDistribution,
+    gradient_function: Callable = finite_differences_gradient,
```

Comment:
Would it make sense to provide deeper type hint? This would definitely help me figure out that `gradient_function` isn't a function of parameters (like `params -> params`) but it's a higher order function (like `(params -> params) -> (params -> params)`)

Reply:
Same for `distance_measure` .

Reply:
Both typehints updated in b8518d1